### PR TITLE
Set ERC-1620 to "Last Call"

### PR DESCRIPTION
**Pull request by PaulRBerg**
As per the announcement in #1620, update the markdown version of ERC-1620 to its final version from Sep 27, 2019.

### CODE REVIEW

**Comment by eip-automerger** [comment]
Hi! I'm a bot, and I wanted to automerge your PR, but couldn't because of the following issue(s):

 - Trying to change EIP 1620 state from Draft to Last Call

**Comment by PaulRBerg** [comment]
Hey @axic! Sorry to tag you, I was wondering if this would be a good time to merge the PR? It seems changing the status to "Last Call" requires a moderator approval.

**Comment by axic** [comment]
@PaulRBerg sorry this was totally missed, can you rebase this if it is still valid?

**Comment by PaulRBerg** [comment]
No worries, just rebased now.

File: EIPS/eip-1620.md
Comment (axic) [inline EIPS/eip-1620.md:11]: If you are still aiming to last call this, let's merge it, but please change the `review-period-end` to be valid, e.g at least 14 days from today.

Comment (PaulRBerg) [reply EIPS/eip-1620.md:11]: Can we make it "Final"?

Comment (axic) [reply EIPS/eip-1620.md]: Yes, after it has passed the actual last call time in the RSS feed. Blame the editors for getting this merged at the time it should have been :(

Comment (PaulRBerg) [reply EIPS/eip-1620.md]: Sorted. Happy to wait.

**Review by axic — APPROVED** [review]

File: EIPS/eip-1620.md
Comment (axic) [inline EIPS/eip-1620.md:44]: I'm not sure this paragraph is relevant here, but will defer to @MicahZoltu's judgement.

File: EIPS/eip-1620.md
Comment (axic) [inline EIPS/eip-1620.md:60]: ERC20 should be properly formatted and be a link, likewise with any other link to the EIP. This can be sorted in this PR or subsequent PRs while in last call.

File: EIPS/eip-1620.md
Comment (axic) [inline EIPS/eip-1620.md:86]: If you are using Solidity syntax here, then I suggest to perhaps consider formatting it according to the style guide for readability, but at least add a semicolon to the end, so it could be copy pasted into an interface.

Furthermore some ERCs chose to describe it as a Solidity interface.

File: EIPS/eip-1620.md
Comment (axic) [inline EIPS/eip-1620.md:167]: ERC-1620 here must be linked.

File: EIPS/eip-1620.md
Comment (axic) [inline EIPS/eip-1620.md:86]: I'm curious if you define a `struct` above, why not return the `struct` in this function? Is that because only ABIEncoderV2 supported it?

**Review by axic — APPROVED** [review]
I think it is fine to go ahead, hoping there will be some feedback from reviewers. I'd ask @MicahZoltu for brief feedback, otherwise I think we should merge this tomorrow.

File: EIPS/eip-1620.md
Comment (PaulRBerg) [reply EIPS/eip-1620.md:86]: Yup. And if this is going to be "Final" soon, I don't want it be non-compliant with the latest Solidity.

**Comment by PaulRBerg** [comment]
Hey @axic, I applied your recommendations, except for this:

> Consider formatting it according to the style guide for readability

How would you rewrite the functions in this PR so that they follow the style guide?

**Comment by axic** [comment]
> > Consider formatting it according to the style guide for readability
>
> How would you rewrite the functions in this PR so that they follow the style guide?

I mean that they are formatted like this:
```solidity
function getStream(uint256 streamId)
  view 
  returns (address sender, address recipient, uint256 deposit, address token, uint256 startTime, uint256 stopTime, uint256 remainingBalance, uint256 ratePerSecond);
```

Which makes is easier to read. I think the style guide has more recommendations.

I'd suggest to check out some ERCs which define a Solidity interface upfront and explain the functions, either as you do now, or via NatSpec.

> As with regard to ERC-165, I'm not particularly familiar with it and I'd be happy to not include it in ERC-1620 for now.

See [ERC-165](https://eips.ethereum.org/EIPS/eip-165) which basically means support of an interface can be determined via checking for the interface id. And defining a Solidity interface makes this very apparent to the reader too.

**Comment by PaulRBerg** [comment]
Thanks for your suggestions! I look forward to improving the ERC-1620 before it is moved to "Final".

But in the meantime, can we move it to "Last Call"? I've been waiting for this for a while haha.

File: EIPS/eip-1620.md
Comment (MicahZoltu) [inline EIPS/eip-1620.md:16]: > Block timestamps are used as an on-chain proxy for time.

This seems like an implementation detail, which is not appropriate for a standardized interface.  If we remove that, we are left with:
> We define money streaming as continuous payments over a finite period of time

Which seems like the definition of a new phrase "money streaming" and an EIP isn't the right place to try to define new terminology.

Consider rewriting this to more clearly define what this is trying to standardize.  If you are just trying to standardize a phrase, then I recommend closing this and pursuing that elsewhere.

File: EIPS/eip-1620.md
Comment (MicahZoltu) [inline EIPS/eip-1620.md:30]: Recommend phrasing this from the point of view of a final EIP.  This means that this isn't "taking a stab at" anything, but instead it is strongly defining something.

File: EIPS/eip-1620.md
Comment (MicahZoltu) [inline EIPS/eip-1620.md:44]:
```suggestion
```
EIPs shouldn't be platforms for advertising specific products.  The list above is reasonable, and more clearly defines for users what this EIP may be useful for.

File: EIPS/eip-1620.md
Comment (MicahZoltu) [inline EIPS/eip-1620.md:42]: Consider a comma separated list rather than a bulleted list as each of these are one-word long so we don't really need the strong separation between them.

File: EIPS/eip-1620.md
Comment (MicahZoltu) [inline EIPS/eip-1620.md:181]: This isn't an official section, and while unofficial sections are not strictly prohibited we try to avoid them in favor of the standardized sections.  Keeping in mind that EIPs are technical specifications, and not meant as a store of history description of use cases, can we just delete this whole section?

**Review by MicahZoltu — CHANGES_REQUESTED** [review]
It is unclear to me why this needs to be a standard rather than just a "good idea".  An interface standard is useful when you have a many-to-many relationship like browsers (many) to websites (many) and when you don't have a fixed set of either one up front (new browsers come and go, new websites come and go).  In this case, each stream arrangement will involve exactly 2 parties, and those parties can agree on using whatever interface/tools they want.  It is unclear why we need a global standard that all money streams need to follow.  Can instead people just use some application like Sablier?

The specification also feels unnecessarily large in the case where we want standardized tooling (e.g., wallets) to support current stream balances.  It seems like a simple:
```solidity
interface MoneyStream {
    function balanceOf(address user) returns (uint256 amount);
}
```
would be sufficient so users can see their current balance of a given money stream.  Perhaps one more function for getting the details of the stream.

A potentially better pattern here may be to create a new contract per stream, rather than having a single contract that encapsulates multiple streams.  Perhaps in the rationale section you could describe why multiple streams per contract is valuable or necessary?

**Comment by PaulRBerg** [comment]
Hey everyone, sorry for not replying here earlier. I read @MicahZoltu's suggestions, started fixing them, but then I realised that what's left is not much.

Micah is right. The specification and its potential reach as a standard are not ample enough to warrant an ERC.

**Comment by edsonayllon** [comment]
@MicahZoltu @axic will this EIP be moved to `Last Call`, or should it be moved to `Withdrawn`?

File: EIPS/eip-1620.md
Comment (edsonayllon) [reply EIPS/eip-1620.md:44]: +1

File: EIPS/eip-1620.md
Comment (edsonayllon) [reply EIPS/eip-1620.md:16]: Well, they have code, so it's not just terminology. Should be something like, "Standard for sending continuous payments over time."

File: EIPS/eip-1620.md
Comment (edsonayllon) [reply EIPS/eip-1620.md:30]: +1. If it is "taking a stab at" this implies the EIP is not functional. Should be rephrased.

File: EIPS/eip-1620.md
Comment (edsonayllon) [reply EIPS/eip-1620.md:181]: Marketing to products in EIPs should be removed. Having references in themselves isn't too bad, especially to avoid plagiarism.

**Comment by MicahZoltu** [comment]
@edsonayllon Ah, this should be moved to `Withdrawn`.  Probably should have done that a while ago.  To any future readers, I'm a huge fan of https://sablier.finance/ which is what @paulrberg went on to build based on this idea.  It has an unfortunate minor security bug but hopefully a future updated contract will fix that.

**Comment by paulrberg** [comment]
Can confirm that I'm okay with this being withdrawn.

Thanks for the shout-out, Micah!

**Pull request closed without merging**
